hasher-matcher-actioner/webapp/src/pages/bank-management/AddBankMemberModal.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ */
+
+import React, {useState} from 'react';
+import axios from 'axios';
+import {Modal, Container, Row, Col, Form, Button} from 'react-bootstrap';
+import {setNestedObjectValues} from 'formik';
+import {ContentType} from '../../utils/constants';
+import BankMemberForm from '../../forms/BankMemberForm';
+import {addBankMember, fetchMediaUploadURL} from '../../Api';
+
+type AddBankMemberModalProps = {
+  bankId: string;
+  bankName: string;
+  show: boolean;
+  contentType: ContentType;
+  onCloseClick: (didAdd: boolean) => void;
+};
+
+function putFileWithProgress(
+  url: string,
+  file: File,
+  progressCallback: (progress: number) => void,
+): Promise<void> {
+  const config = {
+    // Can't type the following line any better. Even axios types leave it as any!
+    onUploadProgress(progressEvent: any) {
+      const percentCompleted = Math.round(
+        (progressEvent.loaded * 100) / progressEvent.total,
+      );
+      progressCallback(percentCompleted);
+    },
+    headers: {
+      'Content-Type': file.type,
+    },
+  };
+
+  return axios.put(url, file, config).then(() => {
+    progressCallback(100);
+  });
+}
+
+export default function AddBankMemberModal({
+  show,
+  bankId,
+  bankName,
+  contentType,
+  onCloseClick,
+}: AddBankMemberModalProps): JSX.Element {
+  let title;
+  switch (contentType) {
+    case ContentType.Photo:
+      title = `Add a Photo to ${bankName}`;
+      break;
+    case ContentType.Video:
+      title = `Add a Video to ${bankName}`;
+      break;
+    default:
+      title = 'Unsupported content type';
+  }
+
+  const [fileUploadProgress, setFileUploadProgress] = useState<number>(0);
+
+  const closeWrapper = (didAdd = false) => {
+    setFileUploadProgress(0);
+    onCloseClick(didAdd);
+  };
+
+  const handleSubmit = (file: File, notes: string) => {
+    // Mark upload in progress:
+    setFileUploadProgress(1);
+
+    const fileNameParts = file.name.split('.');
+    const extension = `.${fileNameParts[fileNameParts.length - 1]}`;
+
+    // Obtain a media URL
```

Comment:
I prefer not to because there are state updates I want to do between those actions. eg. putFileWithProgress would not be possible without passing in a callback. 

This gives the component's author flexibility.